README.md
TV Movie API
============
#### _An objectivly better TV Series and Movie API._

It contains:
 - Metadata about movies (taken from [Trakt](http://trakt.tv/))
 - Metadata about TV Shows and individual episodes (taken from [Trakt](http://trakt.tv/))
 - Multiple quality magnet links for every movie and episode
 - Ability to easily filter content to the users' content

Installation
============
1. Install MongoDB (version `3.0+`)
2. Install NodeJS (version `5.0.0+`)
3. install dependencies `cd tv-movie-api` and `npm install`
4. Start the API using either of the following,
 - `npm start` 
 - `node --harmony --harmony_destructuring --harmony_default_parameters --use_strict index.js`

Routes
======
The API contains the following 'routes' which produce the example output:

`show/:imdb_id` - This returns all info and episodes for a particular show. Useful for the "show details" page in your app.

`movie/:imdb_id` - This returns all info and episodes for a particular movie.

**Example**

`http://<api-url>/show/tt1475582` returns the following:

```
{
  "_id": "tt1475582",
  "imdb_id": "tt1475582",
  "tvdb_id": "176941",
  "title": "Sherlock",
  "year": "2010",
  "slug": "sherlock",
  "synopsis": "Sherlock is a British television crime drama that presents a contemporary adaptation of Sir Arthur Conan Doyle's Sherlock Holmes detective stories. Created by Steven Moffat and Mark Gatiss, it stars Benedict Cumberbatch as Sherlock Holmes and Martin Freeman as Doctor John Watson.",
  "runtime": "90",
  "rating": {
    "hated": 100,
    "loved": 100,
    "votes": 22043,
    "watching": 5,
    "percentage": 93
  },
  "country": "gb",
  "network": "BBC One",
  "air_day": "Sunday",
  "air_time": "20:30",
  "status": "returning series",
  "num_seasons": 3,
  "last_updated": 1458394535176,
  "images": {
    "fanart": "https://walter.trakt.us/images/shows/000/019/792/fanarts/original/605985cad2.jpg",
    "poster": "https://walter.trakt.us/images/shows/000/019/792/posters/original/6311e0343c.jpg",
    "banner": "https://walter.trakt.us/images/shows/000/019/792/banners/original/ca665b6d1d.jpg"
  },
  "__v": 29,
  "episodes": [
    {
      "tvdb_id": 4103396,
      "season": 2,
      "episode": 1,
      "title": "A Scandal in Belgravia",
      "overview": "Compromising photographs and a case of blackmail threaten the very heart of the British establishment, but for Sherlock and John the game is on in more ways than one as they find themselves battling international terrorism, rogue CIA agents, and a secret conspiracy involving the British government.  This case will cast a darker shadow over their lives than they could ever imagine, as the great detective begins a long duel of wits with an antagonist as cold and ruthless and brilliant as himself: to Sherlock Holmes, Irene Adler will always be THE woman.",
      "date_based": false,
      "first_aired": 1325449800,
      "watched": {
        "watched": false
      },
      "torrents": {
        "0": {
          "provider": "EZTV",
          "peers": 0,
          "seeds": 0,
          "url": "magnet:?xt=urn:btih:T6Y4FG35S54U3CWSV2OPAXRQVXHZJ4WV&dn=Sherlock.2x01.A.Scandal.In.Belgravia.HDTV.XviD-FoV&tr=udp://tracker.openbittorrent.com:80&tr=udp://open.demonii.com:80&tr=udp://tracker.coppersurfer.tk:80&tr=udp://tracker.leechers-paradise.org:6969&tr=udp://exodus.desync.com:6969"
        },
        "720p": {
          "provider": "EZTV",
          "peers": 0,
          "seeds": 0,
          "url": "magnet:?xt=urn:btih:GHB4ZITTAO3AMXKNQ4ODBHMFT426NHYU&dn=Sherlock.2x01.A.Scandal.In.Belgravia.720p.HDTV.x264-FoV&tr=udp://tracker.openbittorrent.com:80&tr=udp://open.demonii.com:80&tr=udp://tracker.coppersurfer.tk:80&tr=udp://tracker.leechers-paradise.org:6969&tr=udp://exodus.desync.com:6969"
        },
        "480p": {
          "provider": "EZTV",
          "peers": 0,
          "seeds": 0,
          "url": "magnet:?xt=urn:btih:T6Y4FG35S54U3CWSV2OPAXRQVXHZJ4WV&dn=Sherlock.2x01.A.Scandal.In.Belgravia.HDTV.XviD-FoV&tr=udp://tracker.openbittorrent.com:80&tr=udp://open.demonii.com:80&tr=udp://tracker.coppersurfer.tk:80&tr=udp://tracker.leechers-paradise.org:6969&tr=udp://exodus.desync.com:6969"
        }
      }
    },
    ...
  ],
  "genres": [
    "drama",
    "crime",
    "mystery"
  ]
}
```

`shows/` - Returns the number of pages available to list 50 shows at a time (used for pagination etc).

`movies/` - Returns the number of pages available to list 50 movies at a time.

`shows/:page` - Retuns a list of 50 shows with part of the metadata (For showing whats avalible to watch).

`movies/:page` - Retuns a list of 50 movies with part of the metadata.

**Example**

`http://<api-url>/shows/1`

```
[
  {
    "_id": "tt0944947",
    "imdb_id": "tt0944947",
    "tvdb_id": "121361",
    "title": "Game of Thrones",
    "year": "2011",
    "slug": "game-of-thrones",
    "rating": {
      "hated": 100,
      "loved": 100,
      "votes": 44234,
      "watching": 37,
      "percentage": 94
    },
    "num_seasons": 5,
    "last_updated": 1458401017618,
    "images": {
      "fanart": "https://walter.trakt.us/images/shows/000/001/390/fanarts/original/76d5df8aed.jpg",
      "poster": "https://walter.trakt.us/images/shows/000/001/390/posters/original/93df9cd612.jpg",
      "banner": "https://walter.trakt.us/images/shows/000/001/390/banners/original/9fefff703d.jpg"
    }
  },
  {
    "_id": "tt0903747",
    "imdb_id": "tt0903747",
    "tvdb_id": "81189",
    "title": "Breaking Bad",
    "year": "2008",
    "slug": "breaking-bad",
    "rating": {
      "hated": 100,
      "loved": 100,
      "votes": 38219,
      "watching": 13,
      "percentage": 94
    },
    "num_seasons": 5,
    "last_updated": 1458396988175,
    "images": {
      "fanart": "https://walter.trakt.us/images/shows/000/001/388/fanarts/original/fdbc0cb02d.jpg",
      "poster": "https://walter.trakt.us/images/shows/000/001/388/posters/original/fa39b59954.jpg",
      "banner": "https://walter.trakt.us/images/shows/000/001/388/banners/original/c53872a7e2.jpg"
    }
  },
  {
    "_id": "tt0898266",
    "imdb_id": "tt0898266",
    "tvdb_id": "80379",
    "title": "The Big Bang Theory",
    "year": "2007",
    "slug": "the-big-bang-theory",
    "rating": {
      "hated": 100,
      "loved": 100,
      "votes": 34414,
      "watching": 33,
      "percentage": 85
    },
    "num_seasons": 5,
    "last_updated": 1458401011376,
    "images": {
      "fanart": "https://walter.trakt.us/images/shows/000/001/409/fanarts/original/cff0b01ee7.jpg",
      "poster": "https://walter.trakt.us/images/shows/000/001/409/posters/original/8adfe77938.jpg",
      "banner": "https://walter.trakt.us/images/shows/000/001/409/banners/original/cfd96bef0d.jpg"
    }
  },
  ...
]
```

`shows/select/:ids` - Returns array of shows (matching the ids) with part of the metadata.

`movies/select/:ids` - Returns array of movies (matching the ids) with part of the metadata.

**Example**

`http://<api-url>/shows/select/tt0944947,tt4016454,tt2364582`

```
[
  {
    "_id": "tt3032476",
    "imdb_id": "tt3032476",
    "tvdb_id": "273181",
    "title": "Better Call Saul",
    "year": "2015",
    "slug": "tt3032476",
    "rating": {
      "percentage": 86,
      "watching": 49,
      "votes": 4391,
      "loved": 100,
      "hated": 100
    },
    "num_seasons": 2,
    "images": {
      "banner": "https://walter.trakt.us/images/shows/000/059/660/banners/original/3f00b6cebf.jpg",
      "poster": "https://walter.trakt.us/images/shows/000/059/660/posters/original/a847b27956.jpg",
      "fanart": "https://walter.trakt.us/images/shows/000/059/660/fanarts/original/5885092434.jpg"
    }
  },
  {
    "_id": "tt4532368",
    "imdb_id": "tt4532368",
    "tvdb_id": "295760",
    "title": "DC's Legends of Tomorrow",
    "year": "2016",
    "slug": "dc-s-legends-of-tomorrow",
    "rating": {
      "percentage": 76,
      "watching": 52,
      "votes": 860,
      "loved": 100,
      "hated": 100
    },
    "num_seasons": 1,
    "images": {
      "banner": "https://walter.trakt.us/images/shows/000/098/898/banners/original/fc5fc1bb56.jpg",
      "poster": "https://walter.trakt.us/images/shows/000/098/898/posters/original/7ecb6f08f0.jpg",
      "fanart": "https://walter.trakt.us/images/shows/000/098/898/fanarts/original/cfbbb12db3.jpg"
    }
  },
  ...
]
```

**Sorting**

Most API results can be sorted and filtered with the following query string:

`?sort=`

Possible options are:
- Name: sort by title of the movie/show
- Year: sort by the year the movie/show first aired
- Updated sort by the most recently aired movie/show
- Rating: sort by the highest rated movies/shows
- Trending: sort by which moves/shows are trending

You can change the order of the sort by adding `&order=1` or `&order=-1` to the query string.

**Filtering**

There are 3 ways of filtering content:
 - `?keywords=`: filter content by keywords (for search functionality)
 - `?genre=`: filter content based on genres
 - `?quality=`: **_movies only_** filter content by quality avaliable

These filters can be used separately or together.

Versioning
==========
For transparency and insight into our release cycle, and for striving to maintain backward compatibility, this project will be maintained according to the [Semantic Versioning](http://semver.org/) guidelines as much as possible.

Releases will be numbered with the following format: `<major>.<minor>.<patch>-<build>`

Constructed with the following guidelines:
- A new _major_ release indicates a large change where backwards compatibility is broken.
- A new _minor_ release indicates a normal change that maintains backwards compatibility.
- A new _patch_ release indicates a bugfix or small change which does not affect compatibility.
- A new _build_ release indicates this is a pre-release of the version.

License
=======
If you distribute a copy or make a fork of the project, you have to credit this project as source. This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with this program.  If not, see [http://www.gnu.org/licenses/](http://www.gnu.org/licenses/).

--------------------------------------------------------------------------------

**This project and the distribution of this project is not illegal, nor does it violate _any_ DMCA laws. The use of this project, however, may be illegal in your area. Check your local laws and regulations regarding the use of torrents to watch potentially copyrighted content. The maintainers of this project do not condone the use of this project for anything illegal, in any state, region, country, or planet. _Please use at your own risk_.**

--------------------------------------------------------------------------------

Copyright (c) 2016 - TV Movie API - Released under the [GPL v3 license](LICENSE.txt).


CHANGELOG.md
1.2.1 - Grab the popcorn - 1 April 2016
=======================================

Features:
 - Movies can be filtered to show movies
   that have 1080p or 720p torrents using
   the new `?quality=<quality>` filter
 - Hopefully removed CAM torrents from 
   movies

1.2.0 - Grab the popcorn - 20 March 2016
========================================

Info:
 - Changed name to TV Movie API
  - repo renamed to tv-movie-api
 - Added `imdbMap` to config to correct imdb ids
 - Changed required MongoDB version from `3.2+` 
   to `3.0+`

Bug fixes:
 - Fix some temp files not being written correctly
 
Features:
 - Added Movies to the API
  - scrape movies from KAT.cr with providers,
    like the show scraping
  - New routes to get list of movies or a specific
    movie
 - Enable showing of message time for logging, can
   be set for console, log, none or both


1.1.1 - Fix all the things! - 18 March 2016
===========================================

Info:
 - Required Node.js version is now in the readme
   and in the `package.json`
 - Corrected date in the changelog from last update
 - Corrected version numbering
 - Removed dependency on `Q`
 - Replaced `slug` with `imdb` for getting seasonal 
   metadata from trakt.tv
 - Required NodeJS version was changed to `5.0.0+`
 - Added required MongoDB version to README.md
 - Removed deprecated routes

Bug fixes:
 - Fixed meta not updating properly
 - Fixed shows with some datebased episodes not
   being added to the API
 - Fixed error log containing color codes
 - Removed old TODO messages from the code
 - Some fixes to prevent `ETIMEDOUT`
 - Fixed index not properly showing `Unknown` for
   missing values under some circumstances
 - Use `path.join()` for file paths
 - Fixed new bugs introduced during 1.1.1
   - Fixed API not setting the last updated time correctly
   - Status will now be set to `Idle` after scraping is done
   - Partially fixed issues with MongoDB limitations to sorting
 - Made scraping EZTV faster by merging the `getShowDetails` 
   and `getAllEpisodes` functions
 
Features:
 - Logging can be configured to output to the console,  
   log files, both or to not output at all, set from 
   the `config.js`
   - This can be set for `info`, `warnings` and `errors`
     individually 
 - Resets the log files on each scrape
 - Added `repo` to the index
 - Added time it took to update the API last to the index
 - Added routes for log files to index
   - These only exisit if the have been written too and
     have logging to file enabled for the type of log
	 (`info`, `warning` and `error`) enabled logs show
	 there route on the index, the following

1.1.0 - It's getting colorful! - 10 March 2016
==============================================

Features:
 - Get multiple shows by id at once
 - Choose to scrape KAT.cr, EZTV.ag or both from `config.js`
 - Color console output (Enable/Disable in `config.js`)

1.0.2 - It's time to split. - 9 March 2016
==========================================

Info:
 - Forked from: https://github.com/popcorn-official/popcorn-api
 
Bug fixes:
 - Retry on KAT.cr timeout
 - Updated start command in readme

1.0.1 - What's trending? - 6 March 2016
=======================================

Bug fixes:
 - Sort by trending

1.0.0 - Let's kick some ass! - 1 March 2016
===========================================

Features:
 - Scraping EZTV.ag just like the old API
 - Scraping KAT.cr with different providers
 - Able to add more providers for kat.cr scraping

config.js
const config = {
  master: true,
  ip: 0,
  port: 80,
  workers: 2,
  scrapeTime: "0 0 */6 * * *",
  scrapers: {
    "tv": {
      "eztv": true,
      "kat": true
    },
    "movies": {
      "kat": true
    }
  },
  pageSize: 50,
  serverName: "TV Movie API",
  tempDir: "./tmp",
  logs: {
    "global": {
      "showTime": {
        "console": false,
        "log": true
      }
    },
    "info": {
        "file": "tv-movie-api.log",
        "output": {
          "console": true,
          "log": false
        }
    },
    "warning": {
        "file": "tv-movie-api-warning.log",
        "output": {
          "console": false,
          "log": true
        }
    },
    "error": {
      "file": "tv-movie-api-error.log",
      "output": {
        "console": false,
        "log": true
      }
    },
    "request": {
      "file": "tv-movie-api-request.log",
      "output": {
        "console": false,
        "log": true
      }
    }
  },
  colorOutput: true,
  statusFile: "status.json",
  updatedFile: "lastUpdated.json",
  updateTimeFile: "updateTime.json",
  dbHosts: [ "localhost" ],
  maxWebRequest: 2,
  webRequestTimeout: 6,
  traktKey: "70c43f8f4c0de74a33ac1e66b6067f11d14ad13e33cd4ebd08860ba8be014907",
  eztvMap: {
    "10-oclock-live": "10-o-clock-live",
    "battlestar-galactica": "battlestar-galactica-2003",
    "house-of-cards-2013": "house-of-cards",
    "born-survivor-bear-grylls": "man-vs-wild",
    "black-box": "the-black-box",
    "brooklyn-nine-nine": "brooklyn-ninenine",
    "cracked": "cracked-2013",
    "golden-boy": "golden-boy-2013",
    "hank": "hank-2009",
    "hawaii-five-0-2010": "hawaii-fiveo-2010",
    "legit": "legit-2013",
    "louie": "louie-2010",
    "marvels-agent-carter": "marvel-s-agent-carter",
    "marvels-agents-of-shield": "marvel-s-agents-of-s-h-i-e-l-d",
    "marvels-avengers-assemble": "marvel-s-avengers-assemble",
    "marvels-daredevil": "marvel-s-daredevil",
    "marvels-guardians-of-the-galaxy": "marvel-s-guardians-of-the-galaxy",
    "power-2014": "power",
    "reign": "reign-2013",
    "resurrection-us": "resurrection",
    "scandal-us": "scandal-2012",
    "the-fosters": "the-fosters-2013",
    "the-goldbergs": "the-goldbergs-2013",
    "the-good-guys": "the-good-guys-2010",
    "the-killing": "the-killing-us",
    "the-office": "the-office-us",
    "vikings-us": "vikings"
  },
  imdbMap: {
    "tt0093036": "tt3074694",
    "tt0102517": "tt1657505",
    "tt0264270": "the-late-late-show",
    "tt0288918": "big-brother-s-little-brother",
    "tt0413541": "big-brother-s-big-mouth",
    "tt0432648": "celebrity-fit-club-2005",
    "tt0983782": "thank-god-you-re-here",
    "tt1038686": "dominion",
    "tt1039922": "la-ink",
    "tt1083958": "the-pickup-artist",
    "tt1095158": "the-paper",
    "tt1136131": "cheerleader-u",
    "tt1190536": "black-dynamite",
    "tt1198108": "the-chopping-block-2009",
    "tt1262762": "mark-loves-sharon",
    "tt1277979": "hole-in-the-wall",
    "tt1312022": "time-warp",
    "tt1360544": "paris-hilton-s-british-best-friend",
    "tt1371997": "hows-your-news",
    "tt1380582": "missing",
    "tt1405169": "newswipe-with-charlie-brooker",
    "tt1470539": "you-have-been-watching",
    "tt1536736": "john-safran-s-race-relations",
    "tt1607037": "gordon-s-great-escape",
    "tt1538090": "frankie-neffe",
    "tt1673792": "boston-med",
    "tt1717499": "onion-news-network",
    "tt1724572": "the-suspicions-of-mr-whicher",
    "tt1786826": "get-out-alive-with-bear-grylls",
    "tt1803123": "royal-institution-christmas-lectures",
    "tt1886866": "chemistry-2011",
    "tt1926955": "young-james-herriot",
    "tt2069311": "fry-s-planet-word",
    "tt2091762": "the-celebrity-apprentice-australia",
    "tt1989007": "feasting-on-waves",
    "tt2141186": "derek",
    "tt2151670": "planet-earth-live",
    "tt2171637": "the-story-of-musicals",
    "tt2272367": "loiter-squad",
    "tt2292521": "hooters-dream-girls",
    "tt2297363": "duets-2012",
    "tt2317255": "the-secret-policeman-s-ball",
    "tt2346091": "earthflight",
    "tt2724068": "boston-s-finest",
    "tt2788282": "bikinis-boardwalks",
    "tt2814102": "weed-country",
    "tt2936390": "tt2069449",
    "tt2947494": "brooklyn-da",
    "tt3351392": "long-way-round",
    "tt3394574": "the-great-christmas-light-fight",
    "tt3475768": "duck-quacks-don-t-echo-uk",
    "tt3529910": "victoria-wood-s-nice-cup-of-tea",
    "tt3689584": "satisfaction",
    "tt3713876": "kirstie-s-handmade-christmas",
    "tt3888812": "shannons-legends-of-motorsport",
    "tt4135218": "tt1930315",
    "tt4370492": "tt0106179",
    "tt4453314": "jail-las-vegas",
    "tt4472994": "the-strange-case-of-the-law",
    "tt4563714": "louis-theroux",
    "tt4614532": "walking-through-history",
    "tt4726488": "homes-by-the-sea",
    "tt4883746": "ali-g-rezurection",
    "tt4903026": "topp-country",
    "tt4908908": "carol-klein-s-plant-odysseys",
    "tt4930646": "epic-attractions",
    "tt5006660": "tt5090150",
    "tt5013506": "masterchef-new-zealand",
    "tt5047494": "dash-dolls",
    "tt5072140": "sex-diaries",
    "tt5133970": "building-cars-live",
    "tt5176246": "tiny-house-world",
    "tt5397520": "dark-net",
    "tt5489746": "evil-lives-here"
  },
  katMap: {
    "60-minutes-us": "60-minutes",
    "american-crime": "american-crime-1969",
    "bachelor-live": "the-bachelor-live",
    "ballers-2015": "ballers",
    "big-brother-us": "big-brother-2000",
    "blackish": "black-ish",
    "bobs-burgers": "bob-s-burgers",
    "bordertown-2015": "bordertown-2016",
    "celebrity-big-brother": "celebrity-big-brother-2001",
    "chicago-pd": "chicago-p-d",
    "childrens-hospital-us": "childrens-hospital",
    "cooper-barretts-guide-to-surviving-life": "cooper-barrett-s-guide-to-surviving-life-2016",
    "cosmos-a-space-time-odyssey": "cosmos-a-spacetime-odyssey",
    "dcs-legends-of-tomorrow": "dc-s-legends-of-tomorrow",
    "doll-and-em": "doll-em",
    "gold-rush": "gold-rush-2010",
    "greys-anatomy": "grey-s-anatomy",
    "hawaii-five-0-2010": "hawaii-five-0",
    "heartland-ca": "heartland-2007-ca",
    "hells-kitchen-us": "hell-s-kitchen-2005",
    "house-of-cards-2013": "house-of-cards",
    "how-its-made-dream-cars": "how-it-s-made-dream-cars",
    "how-its-made": "how-it-s-made",
    "intelligence-us": "intelligence-2014",
    "its-always-sunny-in-philadelphia": "it-s-always-sunny-in-philadelphia",
    "james-mays-cars-of-the-people": "james-may-s-cars-of-the-people",
    "jericho-2016": "jericho-1969",
    "kitchen-nightmares-us": "kitchen-nightmares",
    "last-man-standing-us": "last-man-standing-2011",
    "law-and-order-svu": "law-order-special-victims-unit",
    "marvels-agent-carter": "marvel-s-agent-carter",
    "marvels-agents-of-s-h-i-e-l-d": "marvel-s-agents-of-s-h-i-e-l-d",
    "marvels-daredevil": "marvel-s-daredevil",
    "marvels-jessica-jones": "marvel-s-jessica-jones",
    "mike-and-molly": "mike-molly",
    "perception": "perception-2012",
    "power-2014": "power",
    "prey-uk": "prey-2014",
    "proof-us": "proof",
    "reckless": "reckless-2014",
    "resurrection-us": "resurrection-2014",
    "revolution-2012": "revolution",
    "rush-us": "rush-2014",
    "sanctuary-us": "sanctuary",
    "satisfaction-us": "satisfaction-2014",
    "scandal-us": "scandal",
    "schitts-creek": "schitt-s-creek",
    "second-chance": "second-chance-2016",
    "stan-lees-lucky-man": "stan-lee-s-lucky-man",
    "survivors-remorse": "survivor-s-remorse",
    "teen-wolf": "teen-wolf-2011",
    "the-bridge-us": "the-bridge-2013",
    "the-comedians-us": "the-comedians-2015",
    "the-kennedys-uk": "the-kennedys-2015",
    "the-league": "the-league-2009",
    "the-librarians-us": "the-librarians-2014",
    "the-magicians-us": "the-magicians",
    "this-is-england-90": "this-is-england-90-2015",
    "whose-line-is-it-anyway-us": "whose-line-is-it-anyway-1998",
    "young-and-hungry": "young-hungry",
    "youre-the-worst-2014": "you-re-the-worst",
    "youre-the-worst": "you-re-the-worst"
  },
  showProviders: [{
    name: "ZonerLOL",
    query: {
      query: "x264-LOL",
      min_seeds: 3
    }
  }, {
    name: "Zonerw4f",
    query: {
      query: "x264 w4f",
      min_seeds: 1
    }
  }, {
    name: "Zonerfleet",
    query: {
      query: "x264 FLEET"
    }
  }, {
    name: "Zoner720p",
    query: {
      query: "x264 720p",
      min_seeds: 1,
      uploader: "z0n321"
    }
  }, {
    name: "SRIGGA",
    query: {
      query: "x264",
      uploader: "SRIGGA",
      min_seeds: 1
    }
  }, {
    name: "EZTV",
    query: {
      query: "x264",
      uploader: "eztv",
      min_seeds: 1
    }
  }, {
    name: "ZonerHDTV",
    query: {
      query: "x264 HDTV",
      min_seeds: 3,
      uploader: "z0n321"
    }
  }, {
    name: "x264HDTV",
    query: {
      query: "x264 720p HDTV",
      uploader: "z0n321"
    }
  }, {
    name: "Zoner1080p",
    query: {
      query: "x264 1080p",
      uploader: "z0n321"
    }
  }, {
    name: "Zoneravs",
    query: {
      query: "X264-AVS"
    }
  }, {
    name: "ZonerDeflate",
    query: {
      query: "X264-DEFLATE"
    }
  }, {
    name: "Zonerdimension",
    query: {
      query: "X264-Dimension",
      min_seeds: 1
    }
  }, {
    name: "Zoneravs",
    query: {
      query: "X264-AVS"
    }
  }, {
    name: "KILLERS",
    query: {
      query: "x264 KILLERS",
      min_seeds: 3
    }
  }, {
    name: "2HD_x264",
    query: {
      query: "x264-2HD",
      min_seeds: 1
    }
  }, {
    name: "ettv",
    query: {
      query: "x264",
      min_seeds: 3,
      uploader: "ettv"
    }
  }, {
    name: "Brasse0",
    query: {
      uploader: "brasse0"
    }
  }, {
    name: "ETHD",
    query: {
      uploader: "ethd"
    }
  }, {
    name: "Thhaque",
    query: {
      uploader: "Thhaque"
    }
  }, {
    name: "FUM",
    query: {
      query: "X264-FUM"
    }
  }, {
    name: "CRiMSON",
    query: {
      query: "CRiMSON"
    }
  }, {
    name: "DHD",
    query: {
      query: "DHD"
    }
  }, {
    name: "BAJSKORV",
    query: {
      query: "BAJSKORV"
    }
  }, {
    name: "TLA",
    query: {
      query: "TLA"
    }
  }, {
    name: "RiVER",
    query: {
      query: "RiVER"
    }
  }, {
    name: "BATV",
    query: {
      query: "BATV"
    }
  }, {
    name: "VTV",
    query: {
      uploader: "VTV"
    }
  }],
  movieProviders: [{
    name: "YIFY",
    query: {
      query: "720p | 1080p",
      uploader: "YIFY"
    }
  }, {
    name: "SPARKS",
    query: {
      query: "SPARKS"
    }
  }, {
    name: "Megaradon",
    query: {
      query: "720p | 1080p",
      uploader: "megaradon"
    }
  }, {
    name: "Z0n321",
    query: {
      query: "720p | 1080p",
      uploader: "z0n321"
    }
  }]
};

module.exports = config;